src/Database/LSMTree/Internal/IndexOrdinaryAcc.hs

```diff
@@ -64,47 +63,51 @@ new maxKeyCount minChunkSize = assert (maxKeyCount >= 0)      $
                                newSTRef 0                    <*>
                                createBaler minChunkSize
 
-{-
-    Appends a single key to the key list of an index and outputs newly available
-    chunks of the serialised key list.
--}
-appendKey :: SerialisedKey -> IndexOrdinaryAcc s -> ST s [Chunk]
-appendKey key@(SerialisedKey' keyBytes)
-          (IndexOrdinaryAcc buffer keyCountRef baler)
-    = do
-          keyCount <- readSTRef keyCountRef
-          Mutable.write buffer keyCount key
-          writeSTRef keyCountRef $! succ keyCount
-          maybeToList <$> feedBaler [keySizeBytes, keyBytes] baler
-    where
-
-    keySize :: Int
-    !keySize = Primitive.length keyBytes
-
-    keySizeAsWord16 :: Word16
-    !keySizeAsWord16 = assert (keySize <= fromIntegral (maxBound :: Word16)) $
-                       fromIntegral keySize
-
-    keySizeBytes :: Primitive.Vector Word8
-    !keySizeBytes = mkPrimVector 0 2 $
-                    runST $ do
-                        rep <- newByteArray 2
-                        writeByteArray rep 0 keySizeAsWord16
-                        unsafeFreezeByteArray rep
-
 {-|
     Appends keys to the key list of an index and outputs newly available chunks
     of the serialised key list.
 
     __Warning:__ Appending keys whose length cannot be represented by a 16-bit
     word may result in a corrupted serialised key list.
 -}
-append :: Append -> IndexOrdinaryAcc s -> ST s [Chunk]
-append (AppendSinglePage _ key)        index = appendKey key index
-append (AppendMultiPage key pageCount) index = fmap concat                $
-                                               sequence                   $
-                                               genericReplicate pageCount $
-                                               appendKey key index
+append :: Append -> IndexOrdinaryAcc s -> ST s (Maybe Chunk)
+append instruction (IndexOrdinaryAcc buffer keyCountRef baler)
+    = case instruction of
+          AppendSinglePage _ key -> do
+              keyCount <- readSTRef keyCountRef
+              Mutable.write buffer keyCount key
+              writeSTRef keyCountRef $! succ keyCount
+              feedBaler (keyListElem key) baler
+          AppendMultiPage key pageCount -> do
```

Comment:
`pageCount` is not the number of pages, it's the number of *overflow pages*. For `AppendMultiPage _ noverflow`, `pagecount = noverflow + 1`.

Reply:
Oops, you’re right. Fixed in a4fcb497a9d386a2a6aba4e9528b2e27fb4c48de.